pkg/metrics/metrics.go

```diff
@@ -147,3 +160,11 @@ func updateX509Expiry(name, namespace string, cert *x509.Certificate) {
 		"name":      name,
 		"namespace": namespace}).Set(float64(expiryTime.Unix()))
 }
+
+// UpdateCertificateIssuedLatency updates the amount of time for a namespace secret to be issued after a certificate has been created.
+func (m *Metrics) UpdateCertificateIssuedLatency(crt *v1alpha1.Certificate, secret *v1.Secret) {
+	created := crt.ObjectMeta.CreationTimestamp.Time
+	issued := secret.ObjectMeta.CreationTimestamp.Time
```

Comment:
A certificate may be created once, a long time ago. It will then be re-issued numerous times throughout its lifetime.
Additionally, the Secret may be created once and then updated continuously.

Therefore we can't rely on the certificate's creation timestamp (nor the secrets creation timestamp) to calculate how long it takes between initial request and issuance. This requires some other mechanism in order to track the time that an issuance was required, and when the certificate is then re-issued.

Right now we do have a general purpose way of detecting whether a certificate needs to be renewed (see the updateStatus method, which computes whether a Secret is 'up to date' for the Certificate resource).

Perhaps we could record this with something more like:
* When the Ready condition is False, set the metric to be Now()->LastTransitionTime
* When the Ready condition is True, clear the metric

This has the advantage of keeping the latency metric up to date on each call to Sync() (so we get gradually increasing graphs), as well as not relying on CreationTimestamps to calculate the issuance latency.

Does that make sense? I'm not the most well-versed in Prometheus metrics, so there may be something here that I am missing 😄

Reply:
So I attempted to implement int the way mentioned above, however this requires quite a bit of passing around variables. Also, since the latency will be measured in seconds, I'm not sure how much precision would be saved doing it with your implementation described above vs. the very simple implementation that I've added here https://github.com/jetstack/cert-manager/pull/1156/commits/051afec44dfffe02d20664da1054df5921da393e. 

Let me know your thoughts 😄